module/VuFind/src/VuFind/View/Helper/Root/Linkify.php

```diff
@@ -45,32 +45,30 @@
  */
 class Linkify extends AbstractHelper
 {
-    /**
-     * Url highlighter
-     *
-     * @var UrlHighlight
-     */
-    protected $urlHighlight;
-
     /**
      * Constructor
      *
-     * @param UrlHighlight $urlHighlight Url highlighter
+     * @param UrlHighlight $urlHighlight            Url highlighter
+     * @param UrlHighlight $urlHighlightExceptEmail Url highlighter that ignores email addresses
      */
-    public function __construct(UrlHighlight $urlHighlight)
+    public function __construct(protected UrlHighlight $urlHighlight, protected UrlHighlight $urlHighlightExceptEmail)
     {
-        $this->urlHighlight = $urlHighlight;
     }
 
     /**
      * Replace urls and emails by html tags
      *
-     * @param string $string String to linkify (must be HTML-escaped)
+     * @param string $string       String to linkify (must be HTML-escaped)
+     * @param bool   $includeEmail If email addresses should also be linkified
      *
      * @return string
      */
-    public function __invoke(string $string): string
+    public function __invoke(string $string, bool $includeEmail = true): string
     {
-        return $this->urlHighlight->highlightUrls($string);
+        if ($includeEmail) {
+            return $this->urlHighlight->highlightUrls($string);
+        } else {
+            return $this->urlHighlightExceptEmail->highlightUrls($string);
+        }
```

Comment:
Could be simplified to avoid multiple return statements:

```suggestion
        return $includeEmail
            ? $this->urlHighlight->highlightUrls($string)
            : $this->urlHighlightExceptEmail->highlightUrls($string);
```

Reply:
Yes of course, that is much better!